20__building_software/10__basic_building.robot

```diff
@@ -52,6 +52,12 @@ Separated Build Phases (Shadow)
     File Should Exist    foobar/foobar-build/RPMS/foobar-0-1.${DEVICE.arch}.rpm
     [Teardown]    Test Teardown
 
+Template Init
+    [Template]    Build Initialized Project
+    qtquick2app    qmake    ${DEVICE.arch}
+    qtquick2app    cmake    ${DEVICE.arch}
+    qtquick2app-qmlonly    qmake    noarch
+
```

Comment:
We should also test build phase separation and shadow builds with cmake etc. So I would rather turn the existing test cases to templates and test those with all combinations of templates and builders. I would use the "Template" suffix to name the templates:

```
Basic Build
    [Template]    Basic Build Template
    ...

Basic Build (Shadow)
    [Template]    Basic Build (Shadow) Template
    ...
...
*** Keywords ***
...
Basic Build Template
    [Arguments] ...
    [Setup] ...
    ....
    [Teardown] ...
...
```

Reply:
Sure, I can do that.

Reply:
Done. Note: For some reason robot does now allow [Setup] in a template - but luckily it's possible to just use the setup keyword directly.